diamond-kinetics/GenericTable.java

```diff
@@ -0,0 +1,110 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+public class GenericTable<T> {
+    private String name;
+    private Database db;
+
+    public Table(String tName, Database db){
+        name = tName;
+        this.db = db;
+    }
+
+    public String getName(){
+        return name;
+    }
+
+    public void create(String createStatement) throws SQLException {
+        try {
+            db.mConnection.prepareStatement(createStatement).execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void drop() {
+        String statement = "DROP TABLE IF EXISTS " + name + " CASCADE;";
+        try {
+            db.mConnection.prepareStatement(statement).execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public int insert(T item, String insertStatement) throws SQLException {
```

Comment:
Can we make this so that the subclasses know their own `insertStatement` string?

As an end user I don't want to have to know about the `insertStatement` I want to use. I just want to be able to do something like:

```
MessageCRUDService messageSvc = new MessageCRUDService(db);
messageSvc.insert(new Message("Hello World"));
```